src/main/java/net/pms/configuration/PmsConfiguration.java

```diff
@@ -2720,12 +2722,29 @@ public void setIpFilter(String value) {
 		configuration.setProperty(KEY_IP_FILTER, value);
 	}
 
-	public void setPreventsSleep(boolean value) {
-		configuration.setProperty(KEY_PREVENTS_SLEEP, value);
-	}
-
-	public boolean isPreventsSleep() {
-		return getBoolean(KEY_PREVENTS_SLEEP, true);
+	public void setPreventSleep(PreventSleepMode value) {
+		if (value == null) {
+			throw new NullPointerException("value cannot be null");
+		}
+		configuration.setProperty(KEY_PREVENT_SLEEP, value.getValue());
+		PMS.get().getSleepManager().setMode(value);
+	}
+
+	public PreventSleepMode getPreventSleep() {
+		PreventSleepMode sleepMode = null;
+		String value = getString(KEY_PREVENT_SLEEP, null);
+		if (value == null && configuration.containsKey(KEY_OLD_PREVENTS_SLEEP)) {
+			// Backwards compatibility
+				sleepMode =
```

Comment:
@Nadahar cool, I'm glad we agree on that. Yes Go is purposely simple to enable quick adoption - as I understand it was developed by creators of C as a successor to it. I like some of its syntax. I often learn new languages briefly. In particular I like that enforced formatting style and I like that its functions can return more than one value at once unlike Java (you can do it in Java but there's a piece of abstraction like wrapping them in an object/array). It fits with a pattern I'm familiar with at my work which is to often deal with two variables at once in callbacks/promises/awaits, where one is the optional value and one is the optional error.
Anyway there are advantages and disadvantages to all languages whether OO, procedural, functional, etc.